Source/WebKit/Platform/IPC/SharedBufferReference.cpp

```diff
@@ -48,12 +48,13 @@ void SharedBufferReference::encode(Encoder& encoder) const
     if (!m_size)
         return;
 
-    SharedMemory::Handle handle;
-    {
-        auto sharedMemoryBuffer = m_memory ? m_memory : SharedMemory::copyBuffer(*m_buffer);
-        sharedMemoryBuffer->createHandle(handle, SharedMemory::Protection::ReadOnly);
+    auto sharedMemoryBuffer = m_memory ? m_memory : SharedMemory::copyBuffer(*m_buffer);
+    if (auto handle = sharedMemoryBuffer->createHandle(SharedMemory::Protection::ReadOnly))
+        encoder << WTFMove(*handle);
+    else {
+        // FIXME: Copy the buffer if shared memory handle creation fails (or if the buffer is too small to make shared memory worth it).
```

Comment:
I'm not sure this is actually achievable, because even when you're performing a "copy" this is done through a shmem.